client/signup/main.jsx

```diff
@@ -96,12 +96,11 @@ const Signup = React.createClass( {
 
 				if ( timeSinceLoading && timeSinceLoading < MINIMUM_TIME_LOADING_SCREEN_IS_DISPLAYED ) {
 					return delay(
-						this.handleFlowComplete.bind( this, dependencies, destination ),
+						this.handleFlowComplete.bind( this, dependencies, utils.getDestination( destination, dependencies, this.props.flowName ) ),
 						MINIMUM_TIME_LOADING_SCREEN_IS_DISPLAYED - timeSinceLoading
 					);
 				}
-
-				return this.handleFlowComplete( dependencies, destination );
+				return this.handleFlowComplete( dependencies, utils.getDestination( destination, dependencies, this.props.flowName ) );
```

Comment:
Shame we need to manually glue `getDestination` to each occurrence of `handleFlowComplete`. Could we compose higher up or something?